Jenkinsfile

```diff
@@ -18,7 +18,50 @@ pipeline {
                 slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
             }
         }
-        stage('Parallel Build') {
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
+          when {
+            allOf {
+              expression { env.CHANGE_ID != null }
+              expression { env.CHANGE_TARGET != null }
+            }
+          }
+          // TODO DDF-2971 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
+          steps {
+            parallel(
+              linux: {
+                node('linux-large') {
+                  retry(3) {
+                    checkout scm
+                  }
+                  timeout(time: 3, unit: 'HOURS') {
+                    withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '-Xmx8192M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC -Djava.security.egd=file:/dev/./urandom', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                      sh 'mvn clean install -B -T 1C -pl !$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
+                      sh 'mvn install -B -Dmaven.test.redirectTestOutputToFile=true -pl $ITESTS -nsu'
+                    }
+                  }
+                }
+              },
+              windows: {
+                node('proxmox-windows') {
+                  bat 'git config --system core.longpaths true'
+                  retry(3) {
+                    checkout scm
+                  }
+                  timeout(time: 3, unit: 'HOURS') {
+                    withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '-Xmx8192M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                      bat 'mvn clean install -B -T 1C -pl !%ITESTS% -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/%CHANGE_TARGET%'
```

Comment:
@oconnormi IIRC, @lessarderic identified that a quick-build is required prior to performing the GIB enabled build otherwise it fails. I would do a quick compare against https://jenkins.phx.connexta.com/job/DDF-Linux-PR/configure

Reply:
@LinkMJB see above question 😄

Reply:
fixed